From: Andy Gospodarek <gospo@redhat.com>
Date: Fri, 25 Apr 2008 10:38:23 -0400
Subject: [net] ipv6: no addrconf for bonding slaves
Message-id: 20080425143823.GO21367@gospo.usersys.redhat.com
O-Subject: ipv6: no addrconf for bonding slaves
Bugzilla: 236750
RH-Acked-by: Neil Horman <nhorman@redhat.com>

The non-bonding hunk from this upstream patch got missed when I did my
large update for 5.2.

commit c2edacf80e155ef54ae4774379d461b60896bc2e
Author: Jay Vosburgh <fubar@us.ibm.com>
Date:   Mon Jul 9 10:42:47 2007 -0700

    bonding / ipv6: no addrconf for slaves separately from master

This will resolve BZ 236750.

This patch was customer and partner verified.

 net/ipv6/addrconf.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index ab7c344..f0b176a 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -2268,6 +2268,9 @@ static int addrconf_notify(struct notifier_block *this, unsigned long event,
 	switch(event) {
 	case NETDEV_UP:
 	case NETDEV_CHANGE:
+		if (dev->flags & IFF_SLAVE)
+			break;
+
 		if (event == NETDEV_UP) {
 			if (!netif_carrier_ok(dev)) {
 				/* device is not ready yet. */
